Makefile variable issues

The Makefile says

```
# You can set these variables from the command line.
SPHINXOPTS    =
SPHINXBUILD   = python3 -msphinx
SOURCEDIR     = .
BUILDDIR      = _build
```

But if you try to set them, these explicit definitions in the makefile override them.. You have comment them out to have the command line variables work

```
$ SPHINXBUILD="~/maddocs_env/bin/python3 -msphinx" make help
/usr/bin/python3: No module named sphinx
make: *** [Makefile:12: help] Error 1
$ nano Makefile
$ git diff
diff --git a/Makefile b/Makefile
index 3a9183e..115dfd9 100644
--- a/Makefile
+++ b/Makefile
@@ -2,8 +2,8 @@
 #
 
 # You can set these variables from the command line.
-SPHINXOPTS    =
-SPHINXBUILD   = python3 -msphinx
+#SPHINXOPTS    =
+#SPHINXBUILD   = python3 -msphinx
 SOURCEDIR     = .
 BUILDDIR      = _build
 
@@ -24,4 +24,4 @@ clean-auto:
 
 .PHONY: auto
 auto:
-       sphinx-autobuild . _build -H 0.0.0.0
\ No newline at end of file
+       sphinx-autobuild . _build -H 0.0.0.0
$ SPHINXBUILD="~/maddocs_env/bin/python3 -msphinx" make help
Sphinx v3.2.1
Please use `make target' where target is one of
  html        to make standalone HTML files
  dirhtml     to make HTML files named index.html in directories
  singlehtml  to make a single large HTML file
  pickle      to make pickle files
  json        to make JSON files
  htmlhelp    to make HTML files and an HTML help project
  qthelp      to make HTML files and a qthelp project
  devhelp     to make HTML files and a Devhelp project
  epub        to make an epub
  latex       to make LaTeX files, you can set PAPER=a4 or PAPER=letter
  latexpdf    to make LaTeX and PDF files (default pdflatex)
  latexpdfja  to make LaTeX files and run them through platex/dvipdfmx
  text        to make text files
  man         to make manual pages
  texinfo     to make Texinfo files
  info        to make Texinfo files and run them through makeinfo
  gettext     to make PO message catalogs
  changes     to make an overview of all changed/added/deprecated items
  xml         to make Docutils-native XML files
  pseudoxml   to make pseudoxml-XML files for display purposes
  linkcheck   to check all external links for integrity
  doctest     to run all doctests embedded in the documentation (if enabled)
  coverage    to run coverage check of the documentation (if enabled)

$ 
```

Patch incoming

--- a/Makefile
+++ b/Makefile
@@ -2,10 +2,12 @@
 #
 
 # You can set these variables from the command line.
-SPHINXOPTS    =
-SPHINXBUILD   = python3 -msphinx
-SOURCEDIR     = .
-BUILDDIR      = _build
+SPHINXOPTS      ?=
+SPHINXBUILD     ?= python3 -msphinx
+SPHINXAUTOBUILD ?= sphinx-autobuild
+SPHINXAUTOOPTS  ?= -H 0.0.0.0
+SOURCEDIR       ?= .
+BUILDDIR        ?= _build
 
 # Put it first so that "make" without argument is like "make help".
 help:
@@ -20,8 +22,8 @@ help:
 
 .PHONY: clean-auto
 clean-auto:
-	rm -rf _build/*
+	rm -rf "$(BUILDDIR)"/*
 
 .PHONY: auto
 auto:
-	sphinx-autobuild . _build -H 0.0.0.0
+	@$(SPHINXAUTOBUILD) "$(SOURCEDIR)" "$(BUILDDIR)" $(SPHINXAUTOOPTS)

